CHANGELOG.md

```diff
@@ -2,6 +2,45 @@
 
 This document includes a curated changelog for each release. We also publish a changelog as the description of a [GitHub release](https://github.com/nginxinc/nginx-kubernetes-gateway/releases), which, by contrast, is auto-generated and includes links to all PRs that went into the release.
 
+## Release 0.3.0
+
+*April 24, 2023*
+
+This release brings:
+- Extensive validation of Gateway API resources for robustness, security and correctness. See the [validation doc](https://github.com/nginxinc/nginx-kubernetes-gateway/blob/v0.3.0/docs/resource-validation.md)
```

Comment:
this is the following link will work after v0.3.0 tag is added, which will  happen when we actually release. to view the links, replace v0.3.0 with main